MedallionShell/Streams/MergedLinesEnumerable.cs

```diff
@@ -92,6 +110,68 @@ private IEnumerator<string> GetEnumeratorInternal()
             }
         }
 
+#if NETSTANDARD2_0_OR_GREATER
+        private async IAsyncEnumerator<string> GetAsyncEnumeratorInternal()
```

Comment:
This function is 95% repeated with GetEnumeratorInternal. Let's dedup the code.

We can make a private helper method to assist with this:

```
private async Task<string?> NextLineAsync(List<ReaderAndTask> tasks, CancellationToken cancellationToken)
{
    // LOGIC:
    // Debug.Assert(tasks.Count is 0 or 2);
    // First, if tasks is empty, initialize with 2 tasks like we currently do on 118-119
    // Figure out which of the 2 tasks is completed. Remove that task and, if the result is not null, replace it
    // by queueing up the next read. 
    // If the result is not null, return the result. If the result is null instead await the other task and return its result.
    // This is basically what we have on 128-144.
}
```

With this in place, our enumerator method becomes:

```
List<ReaderAndTask> tasks = [];
do
{
      var nextLine = await NextLineAsync(tasks, cancellationToken).ConfigureAwait(false);
      if (nextLine is null) { yield break; } // both readers done
      yield return nextLine;
} while (tasks.Count != 1); // both readers not done

var remaining = tasks[0].Reader;
// code we already have to finish this
```

The non-async iterator should be able to be almost the same; only difference is that it has to GetAwaiter().GetResult() on the NextLineAsync task and the remaining loop uses sync reads.